Bug: Observer crash loop

Checking in with our dogfood cluster this morning, I found a worker and job that have been stuck in a `RUNNING` phase for the last 19 hours.

According to worker logs, the worker is running because it's waiting for the job to complete. No issue there.

The job is considered RUNNING by Brigade, but `kubectl describe` of the corresponding pod says it's in a `NotReady` state. Initially, it was not clear to me why. According to both logs and the output of `kubectl describe`, the job actually seems to have run to completion successfully and exited 0, but k8s is _still_ reporting it as `NotReady`.

The it hit me that k8s considers any stopped pod `NotReady`. So no issue there either.

The issues then are these:

1. Why didn't Brigade's observer component detect that the job completed and change its phase accordingly?

1. Why didn't Brigade's observer component time out a worker and a job that have each been running for 19 hours?

Turns out the observer is in a `CrashLoopBackOff` state.

Logs from the most recent crash reveal this:

```
2021/07/07 14:12:12 Starting Brigade Observer -- version v2.0.0-alpha.5.3 -- commit a12b855
E0707 14:12:12.988823       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 9 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x12fd680, 0x1db1310)
	/go/pkg/mod/k8s.io/apimachinery@v0.18.2/pkg/util/runtime/runtime.go:74 +0xa6
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/pkg/mod/k8s.io/apimachinery@v0.18.2/pkg/util/runtime/runtime.go:48 +0x89
panic(0x12fd680, 0x1db1310)
	/usr/local/go/src/runtime/panic.go:969 +0x1b9
main.(*observer).syncJobPod(0xc0003341c0, 0x1447720, 0xc0002e8400)
	/src/v2/observer/jobs.go:108 +0x97b
k8s.io/client-go/tools/cache.ResourceEventHandlerFuncs.OnAdd(...)
	/go/pkg/mod/k8s.io/client-go@v0.18.2/tools/cache/controller.go:218
k8s.io/client-go/tools/cache.(*processorListener).run.func1()
	/go/pkg/mod/k8s.io/client-go@v0.18.2/tools/cache/shared_informer.go:744 +0xc2
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc000059f60)
	/go/pkg/mod/k8s.io/apimachinery@v0.18.2/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000094f60, 0x15c62a0, 0xc00028ed20, 0xc00000ec01, 0xc00008a720)
	/go/pkg/mod/k8s.io/apimachinery@v0.18.2/pkg/util/wait/wait.go:156 +0xad
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000059f60, 0x3b9aca00, 0x0, 0x1, 0xc00008a720)
	/go/pkg/mod/k8s.io/apimachinery@v0.18.2/pkg/util/wait/wait.go:133 +0xe5
k8s.io/apimachinery/pkg/util/wait.Until(...)
	/go/pkg/mod/k8s.io/apimachinery@v0.18.2/pkg/util/wait/wait.go:90
k8s.io/client-go/tools/cache.(*processorListener).run(0xc00030a100)
	/go/pkg/mod/k8s.io/client-go@v0.18.2/tools/cache/shared_informer.go:738 +0x95
k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1(0xc000312300, 0xc000214800)
	/go/pkg/mod/k8s.io/apimachinery@v0.18.2/pkg/util/wait/wait.go:73 +0x51
created by k8s.io/apimachinery/pkg/util/wait.(*Group).Start
	/go/pkg/mod/k8s.io/apimachinery@v0.18.2/pkg/util/wait/wait.go:71 +0x65
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x120609b]

goroutine 9 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/pkg/mod/k8s.io/apimachinery@v0.18.2/pkg/util/runtime/runtime.go:55 +0x10c
panic(0x12fd680, 0x1db1310)
	/usr/local/go/src/runtime/panic.go:969 +0x1b9
main.(*observer).syncJobPod(0xc0003341c0, 0x1447720, 0xc0002e8400)
	/src/v2/observer/jobs.go:108 +0x97b
k8s.io/client-go/tools/cache.ResourceEventHandlerFuncs.OnAdd(...)
	/go/pkg/mod/k8s.io/client-go@v0.18.2/tools/cache/controller.go:218
k8s.io/client-go/tools/cache.(*processorListener).run.func1()
	/go/pkg/mod/k8s.io/client-go@v0.18.2/tools/cache/shared_informer.go:744 +0xc2
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc000059f60)
	/go/pkg/mod/k8s.io/apimachinery@v0.18.2/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc0000b7f60, 0x15c62a0, 0xc00028ed20, 0xc00000ec01, 0xc00008a720)
	/go/pkg/mod/k8s.io/apimachinery@v0.18.2/pkg/util/wait/wait.go:156 +0xad
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000059f60, 0x3b9aca00, 0x0, 0x1, 0xc00008a720)
	/go/pkg/mod/k8s.io/apimachinery@v0.18.2/pkg/util/wait/wait.go:133 +0xe5
k8s.io/apimachinery/pkg/util/wait.Until(...)
	/go/pkg/mod/k8s.io/apimachinery@v0.18.2/pkg/util/wait/wait.go:90
k8s.io/client-go/tools/cache.(*processorListener).run(0xc00030a100)
	/go/pkg/mod/k8s.io/client-go@v0.18.2/tools/cache/shared_informer.go:738 +0x95
k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1(0xc000312300, 0xc000214800)
	/go/pkg/mod/k8s.io/apimachinery@v0.18.2/pkg/util/wait/wait.go:73 +0x51
created by k8s.io/apimachinery/pkg/util/wait.(*Group).Start
	/go/pkg/mod/k8s.io/apimachinery@v0.18.2/pkg/util/wait/wait.go:71 +0x65
```

--- a/v2/observer/jobs.go
+++ b/v2/observer/jobs.go
@@ -104,8 +104,7 @@ func (o *observer) syncJobPod(obj interface{}) {
 	for _, containerStatus := range pod.Status.ContainerStatuses {
 		if containerStatus.Name == pod.Spec.Containers[0].Name {
 			if containerStatus.State.Terminated != nil {
-				status.Ended =
-					&pod.Status.ContainerStatuses[0].State.Terminated.FinishedAt.Time
+				status.Ended = &containerStatus.State.Terminated.FinishedAt.Time
 			}
 			break
 		}

